boards/common/atxmega/board.c

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright (C) 2018 RWTH Aachen, Josua Arndt <jarndt@ias.rwth-aachen.de>
+ *               2021 Gerson Fernando Budke <nandojve@gmail.com>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     boards_common
+ * @{
+ *
+ * @file
+ * @brief       Common implementations for ATxmega boards
+ *
+ * @author      Josua Arndt <jarndt@ias.rwth-aachen.de>
+ * @author      Gerson Fernando Budke <nandojve@gmail.com>
+ *
+ * @}
+ */
+
+#include "board.h"
+#include "cpu_clock.h"
+#include "cpu.h"
+#include "irq.h"
+#include "periph/gpio.h"
+#include "periph/init.h"
+
+#ifndef CPU_ATXMEGA_CLK_SCALE_INIT
+#define CPU_ATXMEGA_CLK_SCALE_INIT    CPU_ATXMEGA_CLK_SCALE_DIV1
+#endif
+#ifndef CPU_ATXMEGA_BUS_SCALE_INIT
+#define CPU_ATXMEGA_BUS_SCALE_INIT    CPU_ATXMEGA_BUS_SCALE_DIV1_1
+#endif
+
+void led_init(void);
+
+void clk_init(void)
```

Comment:
The cpu_init is in cpu/avr8_common and shared between atmega and atxmega variants.
Any changes here can result in an structural re-work for megas. It will be good to me understand what could be the better move. I don't see problems to move xmega clock_init to cpu/atxmega. I've been thinking to change it to weak to allow board tune.

Reply:
Let's not worry about boards fine-tuning it now. You could provide a `weak` implementation of it in `cpu/avr8_common` instead that does nothing, and override it for ATxmega CPUs. Or define an `CPU_AVR8_HAS_CLOCK_INIT` in `cpu.h` for ATxmega boards. You can even use the shiny new macros to evaluate this in C:

```C
#include "kernel_defines.h"

void cpu_init(void) {
    ...
    extern void clk_init(void);
    if (IS_ACTIVE(CPU_AVR8_HAS_CLOCK_INIT)) {
        clk_init();
    }
    ...
}
```

The `weak` symbol is IMO more elegant, but the flag would result in less ROM being used.